File: 1894/1894.cpp
class Solution {
public:
    int chalkReplacer(vector<int>& chalk, int k) {
        int n = std::end(chalk)-std::begin(chalk);
        long s = 0;
        for(int i=0;i<n;i++) s+=chalk[i];
        k %= s;
        int idx = 0;
        while (true){
            k -= chalk[idx];
            if (k<0) return idx;
            idx ++;
            if (idx==n) idx=0;
        }
    }
};

File: Solutions/1894/1894.py
class Solution:
    def chalkReplacer(self, chalk: List[int], k: int) -> int:
        n = len(chalk)
        s = sum(chalk)
        k = k % s
        idx = 0
        while True:
            k -= chalk[idx]
            if k < 0:
                return idx
            idx += 1
            if idx == n: idx=0
